var/spack/repos/builtin/packages/kentutils/package.py

```diff
@@ -7,35 +7,138 @@
+
+    This bundles a custom version of htslib, but can be overridden with ~htslib.
+    Consider adding the ^mysql+client_only dependency to avoid building all mysql/mariadb.
+    """
 
     homepage = "https://genome.cse.ucsc.edu/"
     url = "https://hgdownload.cse.ucsc.edu/admin/exe/userApps.archive/userApps.v453.src.tgz"
 
-    version("459", sha256="0b6e89a183e6385c713cf010a7aeead9da6626d8d2f78c363a4f1bc56ccccebb")
-    # The above archive only goes back to v305. v302 is left for now but deprecated. Suggest
-    # this is dropped on next update (v302 is from 2014!) and the `requires()` removed.
+    maintainers("teaguesterling")
+
+    version("465", sha256="eef17b1f3182d1d9dc99b5c73a6b0468d5d3bd80470f25d3f7706cc1372e04b0")
+    version("464", sha256="24e20fe68e2a2894d802c87662f69a62f71b3c15fafb2e4d6c3c425c63638bb2")
+    version("460", sha256="b955e56ee880074521ef1ab1371491f47e66dc6fdd93b05328386dd675a635fa")
+    version("455", sha256="e458cadad7c4a5c1b8385edafffa1b29380ac725a0c20535bf5a3bab99fe80db")
+    # This version isn't present in the archive any more
+    # Might be worth changing url to: https://github.com/ucscGenomeBrowser/kent-core/tags/...
     version(
-        "302.1",
-        commit="d8376c5d52a161f2267346ed3dc94b5dce74c2f9",
-        git="https://github.com/ENCODE-DCC/kentUtils.git",
+        "459",
+        sha256="0b6e89a183e6385c713cf010a7aeead9da6626d8d2f78c363a4f1bc56ccccebb",
         deprecated=True,
     )
 
-    depends_on("libpng")
-    depends_on("openssl")
-    depends_on("uuid")
-    depends_on("mariadb")
-    depends_on("zlib-api")
-    depends_on("freetype")
-    depends_on("libiconv")
+    variant("libs", default=True, description="Install jk*.a libraries")
+    variant(
+        "htslib",
+        default=True,
+        description="Build with bundled htslib (using an external htslib may lead to errors)",
+        sticky=True,
+    )
+
+    with default_args(type=("build", "link", "run")):
+        depends_on("libpng")
+        depends_on("openssl")
+        depends_on("uuid")
+        depends_on("mysql-client")
+        depends_on("zlib-api")
+        depends_on("freetype")
+        depends_on("libiconv")
+        depends_on("htslib+pic", when="~htslib+libs")
+
+    # The bgzip.c bug present in other packages is present in kent/src/htslib/bgzf.c
+    # Conflicting line: assert(compressBound(BGZF_BLOCK_SIZE) < BGZF_MAX_BLOCK_SIZE);
+    # We can patch this by removing the assertion, but there are still performance issues
+    # See: https://github.com/samtools/htslib/issues/1257
+    conflicts("zlib-ng")
 
-    requires("%gcc", when="@302.1")
+    # Does not add a link to mysql_config, which is required for compilation
+    conflicts("mariadb-c-client")
+
+    # MySQL pointer/integer conversion issue
+    patch("fix-mysql-options-gcc13.patch", when="%gcc@13: ^mysql")
+    # MySQL build flags improperly states the zlib-api library
+    patch("mysql-zlib-workaround.patch", when="%gcc ^mysql")
```

Comment:
Can you extend the comments with provenance for the patches (assuming they are taken from upstream)?

Reply:
I authored thes, so no upstream. I opened a PR for one, so can afford a reference.

Reply:
Added a link to the former. The later is probably an issue we should open for the `mysql`, `mariadb`, and `postgresql` packages as they all use the same approach to provide linker flags to be used for the client libs (or extension libs).